### PR TITLE
Update Boleto and OXXO specificaitons

### PR DESCRIPTION
**Pull request by mikhail-kuzmiankou-cko**
Changes:
  - Adjust spec generation script to support allOf, anyOf, and oneOf keywords
- Update OXXO and Boleto specifications

### CODE REVIEW

**Comment by philippe-leonhardt-cko** [comment]
Why is the api-reference preview of the branch not working? 👀 
https://api-reference.checkout.com/preview/task/DA-100-update-boleto-and-oxxo-specs

**Comment by mikhail-kuzmiankou-cko** [comment]
> Why is the api-reference preview of the branch not working? 
Na idea, I'll investigate that

**Comment by mikhail-kuzmiankou-cko** [comment]
@philippe-leonhardt-cko I don't know why it doesn't work. Travis logs seem fine. It works locally too.

**Comment by mikhail-kuzmiankou-cko** [comment]
Ah, I figured that out. Try this link: https://api-reference.checkout.com/preview/task/da-100-update-boleto-and-oxxo-specs/

Links are not case-insensitive. The branch name in the URL should be in lower case.

**Comment by philippe-leonhardt-cko** [comment]
Is birthDate not required for either Boleto or OXXO anymore?
Was a required field for both payment methods.

**Comment by mikhail-kuzmiankou-cko** [comment]
Interesting, according to docs, it is optional: 
```
birth_date String
User’s birthdate (DD-MM-YYYY). Optional.
```
https://docs.dlocal.com/api-documentation/payins-api-reference/payments#the-payer-object

This is likely true for all payment methods, provided by dLocal, because they add remarks when some payment methods behave differently: 
```
address Address Object 
User’s address. Only required in India.
```

Is birth_date required for our card dLocal integration?

**Comment by kevin-reece-cko** [comment]
> Interesting, according to docs, it is optional:
> 
> ```
> birth_date String
> User’s birthdate (DD-MM-YYYY). Optional.
> ```
> 
> https://docs.dlocal.com/api-documentation/payins-api-reference/payments#the-payer-object
> 
> This is likely true for all payment methods, provided by dLocal, because they add remarks when some payment methods behave differently:
> 
> ```
> address Address Object 
> User’s address. Only required in India.
> ```
> 
> Is birth_date required for our card dLocal integration?

I seem to remember that they said they were not required for oxxo, but it's worth asking them for each specific new payment method: Boleto/Balotto/RapiPago/Pago Facil

File: spec/components/schemas/Payments/RequestSources/PaymentRequestBoletoSource.yaml [hunk 1]
```diff
@@ -1,22 +1,51 @@
...
+            type: string
+            description: 'Payer''s full name.'
+          email:
+            maxLength: 100
+            type: string
+            description: 'Payer''s email address.'
+          document:
+            maxLength: 100
+            type: string
+            description: 'Payer''s document identifier in Brasil, namely, Cadastro de Pessoas Físicas (CPF) or Cadastro Nacional da Pessoa Jurídica (CNPJ).'
```
Comment (kevin-reece-cko) [inline spec/components/schemas/Payments/RequestSources/PaymentRequestBoletoSource.yaml]:
```suggestion
            description: 'Payer''s document identifier in Brazil, namely, Cadastro de Pessoas Físicas (CPF) or Cadastro Nacional da Pessoa Jurídica (CNPJ).'
```

File: spec/components/schemas/Payments/ResponseSources/PaymentResponseBoletoSource.yaml [hunk 2]
```diff
@@ -1,4 +1,99 @@
...
+              - third_party_amount_exceeded
+              - third_party_timeout
+              - third_party_error
+            type: string
+            description: 'Failure code'
+            example: third_party_rejected
+      -
+        type: object
+        properties:
+          integration_type:
```
Comment (kevin-reece-cko) [inline spec/components/schemas/Payments/ResponseSources/PaymentResponseBoletoSource.yaml]: Is this in the response? I don't see it in dlocal source. Is it on a branch?

And do we need it? It seems like duplication of the existence of ticket and redirect url

File: scripts/sync-generated-specs.js [hunk 3]
```diff
@@ -55,23 +57,61 @@ var sync = function(paymentSpecUrl, buildPaymentResponseFunction, outputFilePath
   request({ url: paymentSpecUrl, json: true}, thenBuildOutputAndWrite(buildOutputFunction, outputFilePath));
 };
 
+var processObjectSchema = function(outputSchema, inputSchema) {
```
Comment (kevin-reece-cko) [inline scripts/sync-generated-specs.js]: Can you add a comment or make the method name more descriptive.

File: spec/components/schemas/Payments/ResponseSources/PaymentResponseBoletoSource.yaml (hunk 2 above)
Comment (kevin-reece-cko) [reply spec/components/schemas/Payments/ResponseSources/PaymentResponseBoletoSource.yaml]: Ah yes I remember now, we do need it for the case of sync failure where there is no ticket or redirect url. 

But it's not implemented yet, right?

File: spec/components/schemas/Payments/RequestSources/PaymentRequestBoletoSource.yaml (hunk 1 above)
Comment (mikhail-kuzmiankou-cko) [reply spec/components/schemas/Payments/RequestSources/PaymentRequestBoletoSource.yaml]: Fixed here and in dLocal repo

File: spec/components/schemas/Payments/ResponseSources/PaymentResponseBoletoSource.yaml (hunk 2 above)
Comment (mikhail-kuzmiankou-cko) [reply spec/components/schemas/Payments/ResponseSources/PaymentResponseBoletoSource.yaml]: True, it's not implemented. I've applied necessary changes in dLocal's repository.

File: scripts/sync-generated-specs.js (hunk 3 above)
Comment (mikhail-kuzmiankou-cko) [reply scripts/sync-generated-specs.js]: Yeah, the name is too generic. Renamed!

**Pull request merged**